File: language-support/js/daml-ledger/src/mapping/completion_stream_request.ts
// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

import * as mapping from '.';
import * as ledger from '..';
import * as grpc from 'daml-grpc';

export const CompletionStreamRequest: mapping.Mapping<grpc.CompletionStreamRequest, ledger.CompletionStreamRequest> = {
    toObject(message: grpc.CompletionStreamRequest): ledger.CompletionStreamRequest {
        return {
            applicationId: message.getApplicationId(),
            offset: mapping.LedgerOffset.toObject(message.getOffset()!),
            parties: message.getPartiesList()
        };
    },
    toMessage(object: ledger.CompletionStreamRequest): grpc.CompletionStreamRequest {
        const result = new grpc.CompletionStreamRequest();
        result.setApplicationId(object.applicationId);
        result.setOffset(mapping.LedgerOffset.toMessage(object.offset));
        result.setPartiesList(object.parties);
        return result;
    }
}

File: sdk/canton/community/bindings-java/src/main/java/com/daml/ledger/javaapi/data/CompletionStreamRequest.java
// Copyright (c) 2025 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

package com.daml.ledger.javaapi.data;

import com.daml.ledger.api.v2.CommandCompletionServiceOuterClass;
import org.checkerframework.checker.nullness.qual.NonNull;

import java.util.List;
import java.util.Objects;
import java.util.Optional;

public final class CompletionStreamRequest {

  @NonNull private final String userId;

  @NonNull private final List<@NonNull String> parties;

  @NonNull private final Long beginExclusive;

  public CompletionStreamRequest(
      @NonNull String userId,
      @NonNull List<@NonNull String> parties,
      @NonNull Long beginExclusive) {
    this.userId = userId;
    this.parties = List.copyOf(parties);
    this.beginExclusive = beginExclusive;
  }

  @NonNull
  public String getUserId() {
    return userId;
  }

  @NonNull
  public List<@NonNull String> getParties() {
    return parties;
  }

  public Long getBeginExclusive() {
    return beginExclusive;
  }

  public static CompletionStreamRequest fromProto(
      CommandCompletionServiceOuterClass.CompletionStreamRequest request) {
    return new CompletionStreamRequest(
        request.getUserId(), request.getPartiesList(), request.getBeginExclusive());
  }

  public CommandCompletionServiceOuterClass.CompletionStreamRequest toProto() {
    CommandCompletionServiceOuterClass.CompletionStreamRequest.Builder builder =
        CommandCompletionServiceOuterClass.CompletionStreamRequest.newBuilder()
            .setUserId(userId)
            .addAllParties(parties)
            .setBeginExclusive(beginExclusive);

    return builder.build();
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    CompletionStreamRequest that = (CompletionStreamRequest) o;
    return Objects.equals(userId, that.userId)
        && Objects.equals(parties, that.parties)
        && Objects.equals(beginExclusive, that.beginExclusive);
  }

  @Override
  public int hashCode() {

    return Objects.hash(userId, parties, beginExclusive);
  }

  @Override
  public String toString() {
    return "CompletionStreamRequest{"
        + "userId="
        + userId
        + ", parties="
        + parties
        + ", beginExclusive="
        + beginExclusive
        + '}';
  }
}
